src/main/java/br/unb/cic/js/walker/RepositoryWalkerTask.java

```diff
@@ -12,51 +12,47 @@
 import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 @Builder
 public class RepositoryWalkerTask implements Runnable {
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
-    // The report is the csv file generated by the program after all observations of a given project are finished.
     public final Path output;
-
     public final BufferedWriter results;
-
     public final RepositoryWalker walker;
-
     public final Interval interval;
-
-    // Threads represents the number of visitors working concurrently on a given revision
     public final int threads;
-
     public final int steps;
 
     @Override
     public void run() {
-        // build report and report errors file
         val reportFile = Paths.get(output.toString(), walker.project + ".csv");
         val reportErrors = Paths.get(output.toString(), walker.project + "-errors.txt");
 
-        val content = new StringBuilder();
-        val errors = new StringBuilder();
-
         try {
-            val summaries = walker.traverse(interval.begin, interval.end, steps, threads);
-            
-            summaries.forEach(s -> {
-                        content.append(s.values())
-                       .append("\n");
+            ConcurrentLinkedQueue<Summary> summaries = new ConcurrentLinkedQueue<>();
```

Comment:
Another thing I'd mention is that you don't actually need a ConcurrentLinkedQueue over here since the summaries are already complete and that's the only step running concurrently when the projects threads are set to 1.